packages/plugins/Filter/src/store/index.ts

```diff
@@ -0,0 +1,189 @@
+import Vue from 'vue'
+import {
+  generateSimpleGetters,
+  generateSimpleMutations,
+} from '@repositoryname/vuex-generators'
+import {
+  FilterConfigurationTimeOption,
+  FilterConfiguration,
+  PolarModule,
+} from '@polar/lib-custom-types'
+import ChooseTimeFrame from '../components/ChooseTimeFrame.vue'
+import { FilterGetters, FilterState } from '../types'
+import { updateFeatureVisibility } from '../utils/updateFeatureVisibility'
+
+const getInitialState = (): FilterState => ({
+  category: {},
+  time: {},
+})
+
+const arrayOnlyContains = (array, value) =>
+  array.reduce((accumulator, current) => accumulator && current === value, true)
+
+const setState = (state, path, value) => {
```

Comment:
As this is a utility function as part of the filter plugin, will there ever be a different object other than the FilterState?

Also, if the type of `value` can be arbitrary, then please add `unknown` instead of leaving the type empty.

Reply:
>As this is a utility function as part of the filter plugin, will there ever be a different object other than the FilterState?

It already is since this is a recursive function.

>Also, if the type of value can be arbitrary, then please add unknown instead of leaving the type empty.

https://github.com/Dataport/polar/pull/34/commits/55fd36cea04113474f01c16c02b84a23c3fe8ecd